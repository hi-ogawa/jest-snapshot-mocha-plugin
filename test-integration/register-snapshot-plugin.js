const process = require("process");
const { JestSnapshotMochaPlugin } = require("../build/tsc/src");

const plugin = new JestSnapshotMochaPlugin({
  updateSnapshot: !!process.env.UPDATE_SNAPSHOT,
});

global.matchSnapshot = plugin.matchSnapshot;
exports.mochaHooks = plugin.mochaHooks;
