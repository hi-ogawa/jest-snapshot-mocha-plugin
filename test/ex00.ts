import { describe, it } from "mocha";
import { matchSnapshot } from "./register-snapshot-plugin";

describe("ex00", () => {
  it("integer", () => {
    matchSnapshot(1);
    matchSnapshot(2);
    matchSnapshot(3);
  });

  it("object", () => {
    matchSnapshot({
      x: 1,
      y: 2,
      z: 3,
    });

    matchSnapshot({
      x: 1,
      y: 2,
      z: {
        u: 3,
        v: 4,
      },
    });
  });
});
