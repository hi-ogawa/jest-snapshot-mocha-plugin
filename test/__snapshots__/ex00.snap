// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ex00 integer 1`] = `1`;

exports[`ex00 integer 2`] = `2`;

exports[`ex00 integer 3`] = `3`;

exports[`ex00 object 1`] = `
Object {
  "x": 1,
  "y": 2,
  "z": 3,
}
`;

exports[`ex00 object 2`] = `
Object {
  "x": 1,
  "y": 2,
  "z": Object {
    "u": 3,
    "v": 4,
  },
}
`;
